### DISCUSSION TITLE
Question: network with predictible hostnames

### DISCUSSION THREAD

**Discussion by vsoch** [lima-vm/lima#2018] [Q&A] [answered]
I'm bringing up multiple VMs, and adding them to the same user network:

```
limactl start --network=lima:user-v2 --name=flux-0 ./flux-usernetes.yaml
limactl start --network=lima:user-v2 --name=flux-1 ./flux-usernetes.yaml
```

Looking at the instructions here https://lima-vm.io/docs/config/network/#lima-user-v2-network that seems to be the last option. What I'm able to do is get the actual ip address from `ip a` and then update /etc/hosts of each, adding the hostnames:

```
192.168.104.6  lima-flux-0
192.168.104.7  lima-flux-1
```
And then my application can query for lima-flux-0 or lima-flux-1 and it's associated with an address. I'm wondering if there is a way to automate this, so I could ping lima-flux-0 and get that address without needing to modify /etc/hosts? Note that the hostname command returns the correct name in the VM, but I can't use that to ping another one (without the addition. Thank you!

**Comment by AkihiroSuda** [comment]
We should enhance the built-in DNS to support looking up hostnames.

A workaround would be to use mDNS, but I couldn't get it to work.

**Comment by balajiv113** [comment, marked as answer]
> then update /etc/hosts of each

For user-v2 it should already work. If you call `lima-flux-0.internal` from lima-flux-1 instance. 

Just that you need to user .internal suffix for it to work. I remember this was documented somewhere i will check and add if this is not captured

**Reply by vsoch** [reply]
Ah, this is fantastic! I'll try this out this morning and report back.

**Reply by a-h** [reply]
Great! Do you know if there's a way to access the Lima VMs from the host by their DNS entries on Linux?

I've ran `limactl create --name=machine --network=lima:user-v2 --memory=2 --cpus=1 --plain --tty=false template://rocky-8` but from the host, I can't ping:

* machine.local.internal
* machine.internal
* machine
* lima-machine.local.internal
* lima-machine.internal
* lima-machine

**Reply by a-h** [reply]
Oh, I should mention, on Linux. I've tried socket_vmnet and vzNAT, and neither work on Linux.

**Reply by AkihiroSuda** [reply]
`sshuttle --dns --auto-nets --ssh-cmd "ssh -F $HOME/.lima/vm0/ssh.config" --remote lima-vm0` seemed to work, but also seems to break looking up Internet domains (`nslookup google.com`).
This may work stably with some additional options.

**Reply by vsoch** [reply]
It worked great for me! I'm still working on my changes but I'll push (and link) soon. I'm on ubuntu 22.04 and i used the user-v2 network, and I could ping lima-flux-0 from lima-flux-1 and vice versa without adding anything to `/etc/hosts`

**Reply by vsoch** [reply]
okay here is the current configuration(s) etc. for what I'm working on (and adding .internal works to bootstrap my VM cluster) https://github.com/converged-computing/flux-lima/tree/main/usernetes#sanity-checks. This is a WIP so it will likely be changing vs. not!

**Reply by AkihiroSuda** [reply]
Opened a PR for docs:
- https://github.com/lima-vm/lima/pull/2156

**Reply by fwilhe2** [reply]
Wasn't aware of the user-v2 networking mode, this seems super useful to me. Thanks everyone for building/documenting it! :rocket:
